service/src/test/scala/za/co/absa/loginsvc/rest/service/JWTServiceTest.scala

```diff
@@ -115,4 +117,23 @@ class JWTServiceTest extends AnyFlatSpec {
     assert(actualGroups === userWithGroups.groups)
   }
 
+  behavior of "jwks"
+
+  it should "return a JWK that is equivalent to the `publicKey`" in {
+    import scala.collection.JavaConverters._
+
+    val publicKey = jwtService.publicKey
+    val jwks = jwtService.jwks
+    val rsaKey = jwks.getKeys.asScala.head.toRSAKey
+
+    assert(publicKey == rsaKey.toPublicKey)
+  }
+
+  it should "return a JWK with parameters" in {
+    import scala.collection.JavaConverters._
+
+    val jwk = jwtService.jwks.getKeys.asScala.head
```

Comment:
```suggestion
    val keys = jwtService.jwks.getKeys.asScala
    assert(keys.length == 1, "One JWK is expected to be generated now")

    val jwk = keys.head
```

Perhaps we could assert that there is only 1 key now to be checked. Should the set get extended at some point, it would be nice for test to fail to force its adjustment 😈